qutebrowser/config.org
#+TITLE: Qutebrowser Configiration File
#+DESCRIPTION: Max Masterton's Config file for the QuteBrowser
#+PROPERTY: header-args :tangle ~/.config/qutebrowser/config.py
#+AUTHOR: Max Masterton

* Config
** Imports
Here are the imports that are required for the rest of the config:

#+BEGIN_SRC python
import dracula.draw
#+END_SRC

** Dracula Theme
This enables the dracula theme which can be found at https://draculatheme.com

#+BEGIN_SRC python
config.load_autoconfig()
dracula.draw.blood(c,{
    'spacing':{
        'vertical':6,
        'horizontal':8
    }
})
#+END_SRC

** Aliases
This enables shortcuts which speed some things up.

#+BEGIN_SRC python
# c.aliases = {
#     'q' : 'quit',
#     'w' : 'session-save',
#     'wq': 'quit --save'
#     }
#+END_SRC

** Cookies
This automatically enables cookies when they are requested by a website for both chrome and webkit.

#+BEGIN_SRC python
config.set("content.cookies.accept", "all", "chrome-devtools://*")
config.set("content.cookies.accept", "all", "devtools://*")
#+END_SRC


emacs/.doom.d/config.org
#+TITLE: Doom Emacs Config
#+DESCRIPTION: Max Masterton's configiration file for Doom Emacs, An Emacs Framework for the stubborn martian vimmer.
#+PROPERTY: header-args :tangle ~/.doom.d/config.el
#+AUTHOR: Max Masterton

* Notes
+ Control-C Control-C to refresh 
+ M-x org-babel-tangle to tangle to config.el

#+BEGIN_SRC emacs-lisp
;   ___ _ __ ___   __ _  ___ ___
;  / _ \ '_ ` _ \ / _` |/ __/ __|
; |  __/ | | | | | (_| | (__\-- \
;  \___|_| |_| |_|\__,_|\___|___/
;
#+END_SRC

* The Preambles
This code in this section codes for default behaviours from doom emacs

** Name and Email
#+BEGIN_SRC emacs-lisp
(setq user-full-name "Max Masterton"
      user-mail-address "me@maxmasterton.tech"
      ;user-login-name "MaxTheMasterton"
      )
#+END_SRC

** Rudimentary Settings
#+BEGIN_SRC emacs-lisp
(setq-default
 delete-by-moving-to-trash t ; Delete files to trash
 tab-width 4 ; Set tab width
 uniquify-buffer-name-style 'forward ; Uniquify buffer names
 window-combination-resize t ; Take new window space from all existing windows
 x-stretch-cursor t ; Stretch cursor to cover long glyphs
 )

(setq undo-limit 80000000 ; Set undo limit to 80mb
      evil-want-fine-undo t ; More granular changes in insert mode
      auto-save-default t ; Auto save by default
      inhibit-compacting-font-caches t ; Load glyphs into memory
      truncate-string-ellipsis "..." ; Use unicode eliipis instead of "..."
      )

(delete-selection-mode 1) ; Replace selection when inserting text
(global-subword-mode 1) ; Iterate through CamelCase Words with evil bindings
#+END_SRC

* Aesthetic Settings
These settings code for asethetic changes

** Fonts
Doom exposes five (optional) variables for controlling fonts in Doom. Here are the 3 important ones:
 + 'doom-font'
 + 'doom-variable-pitch-font'
 + 'doom-big-font' used for presentations and streaming

They all accept a font-spec, font-string ("Input Mono-12") or xldf font string, you generally only need these two:
#+BEGIN_SRC emacs-lisp
(setq doom-font (font-spec :family "JetBrains Mono" :size 13)
      doom-big-font (font-spec :family "JetBrains Mono" :size 30)
      ;doom-variable-pitch font
      ;doom-serif-font
      )
#+END_SRC

*** Proportional Fonts
#+BEGIN_SRC emacs-lisp
(defun my-buffer-face-mode-variable ()
  "Set font to a variable width (proportional) fonts in the current buffer"
  (interactive)
  (setq buffer-face-mode-face '(family: "JetBrains Mono" :height 100))
  (buffer-face-mode))
(add-hook 'org-mode-hook 'my-buffer-face-mode-variable)
#+END_SRC

** Theme
There are two ways to load a theme. Both assume that the theme is installed and avaliable. You can either set 'doom-theme' or manually load a theme with the 'load-theme option'. I like dracula because the colours stand out and the syntax is consistent.

#+BEGIN_SRC emacs-lisp
(setq doom-theme 'doom-dracula)

;; Flagship themes
;(setq doom-theme 'doom-one
;      doom-theme 'doom-vibrant)
#+END_SRC

** Transparency
The following segment enables transparency, this should only be done if a strong blur is being applied by the compositor (E.g. picom's 'dual-kawase'). The first option is for selected blur the second for idle blur.

#+BEGIN_SRC emacs-lisp
;(set-frame-parameter (selected-frame) 'alpha '(85 . 85))
;(add-to-list 'default-frame-alist '(alpha . (85 . 85)))
#+END_SRC

* Modeline
** Doom Modeline
The Doom Modeline Can also be customized by running 'M-x customize-group RET doom-modeline RET'

#+BEGIN_SRC emacs-lisp
(setq doom-modeline-height 25
      doom-modeline-bar-width 3)
;
(setq doom-modeline-icon (display-graphic-p)
      doom-modeline-buffer-notification-icon t
;      doom-modeline-enable-word-count t ; Enables word count for selected area
      doom-modeline-indent-info t
      doom-modeline-workspace-name t
      )
;
(setq doom-modeline-continuous-word-count-modes '(markdown-mode gfm-mode org-mode))
#+END_SRC

** Standard Modeline
Items visable on the standard GNU modeline are also made visable on the doom modeline

#+BEGIN_SRC emacs-lisp
(display-time-mode 1)
(global-display-line-numbers-mode)
#+END_SRC

** Modeline Buffer Encoding
Almost everything that I edit is encoded in UTF-8 so it's only worth showing buffer encoding on the modeline if it's something other than UTF-8.

#+BEGIN_SRC emacs-lisp
(defun doom-modeline-conditional-buffer-encoding ()
  (setq-local doom-modeline-buffer-encoding
              (unless (or (eq buffer-file-coding-system 'utf-8-unix)
                          (eq buffer-file-coding-system 'utf-8)))))
(add-hook 'after-change-major-mode-hook #'doom-modeline-conditional-buffer-encoding)
#+END_SRC

* Other
** Org Directory
if you use 'org' and you don't want you org files in the default location below, change 'org-directory'. It must be set before org loads! Mine lives in a dedicated org directory

#+BEGIN_SRC emacs-lisp
(setq org-directory "~/org/")
#+END_SRC

* Packages
** Mu4e
Setting up mu4e which is an email client that works within emacs. mu4e and mbysnc must also be installed via the AUR.

#+BEGIN_SRC emacs-lisp
(add-to-list 'load-path "/usr/local/share/emacs/site-lisp/mu4e")
(setq mu4e-get-mail-command "mbsync -c ~/.config/mu4e/mbsyncrc -a"
      mu4e-update-insterval 300
      mu4e-main-buffer-hide-personal-addresses t
      message-send-mail-function 'smtpmail-send-it
      starttls-use-gnutls t
      smtpmail-starttls-credentials '(("smtp.maxmasterton.tech" 587 nil nil))
      mu4e-sent-folder "/account-1/Sent"
      mu4e-drafts-folder "/account-1/Drafts"
      mu4e-trash-folder "/account-1/Trash"
      mu4e-maildir-shortcuts
      '(("/account-1/Inbox"      . ?i)
        ("/account-1/Sent Items" . ?s)
        ("/account-1/Drafts"     . ?d)
        ("/account-1/Trash"      . ?t)))
#+END_SRC


emacs/.doom.d/config.el
;   ___ _ __ ___   __ _  ___ ___
;  / _ \ '_ ` _ \ / _` |/ __/ __|
; |  __/ | | | | | (_| | (__\-- \
;  \___|_| |_| |_|\__,_|\___|___/
;

(setq user-full-name "Max Masterton"
      user-mail-address "me@maxmasterton.tech"
      ;user-login-name "MaxTheMasterton"
      )

(setq-default
 delete-by-moving-to-trash t ; Delete files to trash
 tab-width 4 ; Set tab width
 uniquify-buffer-name-style 'forward ; Uniquify buffer names
 window-combination-resize t ; Take new window space from all existing windows
 x-stretch-cursor t ; Stretch cursor to cover long glyphs
 )

(setq undo-limit 80000000 ; Set undo limit to 80mb
      evil-want-fine-undo t ; More granular changes in insert mode
      auto-save-default t ; Auto save by default
      inhibit-compacting-font-caches t ; Load glyphs into memory
      truncate-string-ellipsis "..." ; Use unicode eliipis instead of "..."
      )

(delete-selection-mode 1) ; Replace selection when inserting text
(global-subword-mode 1) ; Iterate through CamelCase Words with evil bindings

(setq doom-font (font-spec :family "JetBrains Mono" :size 13)
      doom-big-font (font-spec :family "JetBrains Mono" :size 30)
      ;doom-variable-pitch font
      ;doom-serif-font
      )

(defun my-buffer-face-mode-variable ()
  "Set font to a variable width (proportional) fonts in the current buffer"
  (interactive)
  (setq buffer-face-mode-face '(family: "JetBrains Mono" :height 100))
  (buffer-face-mode))
(add-hook 'org-mode-hook 'my-buffer-face-mode-variable)

(setq doom-theme 'doom-dracula)

;; Flagship themes
;(setq doom-theme 'doom-one
;      doom-theme 'doom-vibrant)

;(set-frame-parameter (selected-frame) 'alpha '(85 . 85))
;(add-to-list 'default-frame-alist '(alpha . (85 . 85)))

(setq doom-modeline-height 25
      doom-modeline-bar-width 3)
;
(setq doom-modeline-icon (display-graphic-p)
      doom-modeline-buffer-notification-icon t
;      doom-modeline-enable-word-count t ; Enables word count for selected area
      doom-modeline-indent-info t
      doom-modeline-workspace-name t
      )
;
(setq doom-modeline-continuous-word-count-modes '(markdown-mode gfm-mode org-mode))

(display-time-mode 1)
(global-display-line-numbers-mode)

(defun doom-modeline-conditional-buffer-encoding ()
  (setq-local doom-modeline-buffer-encoding
              (unless (or (eq buffer-file-coding-system 'utf-8-unix)
                          (eq buffer-file-coding-system 'utf-8)))))
(add-hook 'after-change-major-mode-hook #'doom-modeline-conditional-buffer-encoding)

(setq org-directory "~/org/")

(add-to-list 'load-path "/usr/local/share/emacs/site-lisp/mu4e")
(setq mu4e-get-mail-command "mbsync -c ~/.config/mu4e/mbsyncrc -a"
      mu4e-update-insterval 300
      mu4e-main-buffer-hide-personal-addresses t
      message-send-mail-function 'smtpmail-send-it
      starttls-use-gnutls t
      smtpmail-starttls-credentials '(("smtp.maxmasterton.tech" 587 nil nil))
      mu4e-sent-folder "/account-1/Sent"
      mu4e-drafts-folder "/account-1/Drafts"
      mu4e-trash-folder "/account-1/Trash"
      mu4e-maildir-shortcuts
      '(("/account-1/Inbox"      . ?i)
        ("/account-1/Sent Items" . ?s)
        ("/account-1/Drafts"     . ?d)
        ("/account-1/Trash"      . ?t)))
